llvm/test/Transforms/InstCombine/alloca-in-non-alloca-as-sentinel.ll

```diff
@@ -0,0 +1,65 @@
+; NOTE: Assertions have been autogenerated by utils/update_test_checks.py
+; RUN: opt < %s -passes=instcombine -S | FileCheck %s
+
+; Gracefully handle the alloca that is not in the alloca AS (=5)
+
+target datalayout = "e-p:64:64-p1:64:64-p2:32:32-p3:32:32-p4:64:64-p5:32:32-p6:32:32-p7:160:256:256:32-p8:128:128-i64:64-v16:16-v24:32-v32:32-v48:64-v96:128-v192:256-v256:256-v512:512-v1024:1024-v2048:2048-n32:64-z0:1-z2:neg1-z3:neg1-z5:neg1-S32-A5-G1-ni:7:8:9"
+target triple = "amdgcn-amd-amdhsa"
+
+declare void @use(ptr)
+declare void @use2(ptr, ptr)
+
+define weak amdgpu_kernel void @__omp_offloading_802_ea0109_main_l8(ptr %a) {
+; CHECK-LABEL: @__omp_offloading_802_ea0109_main_l8(
+; CHECK-NEXT:  .master:
+; CHECK-NEXT:    [[TMP0:%.*]] = alloca [8 x i8], align 1
+; CHECK-NEXT:    call void @use2(ptr nonnull [[TMP0]], ptr nonnull [[TMP0]])
+; CHECK-NEXT:    ret void
+;
+.master:
+  %0 = alloca i8, i64 8, align 1
+  store ptr undef, ptr %0, align 8
+  call void @use2(ptr %0, ptr %0)
+  ret void
+}
+
+%struct.widget = type { [8 x i8] }
+
+define void @spam(ptr %arg1) {
+; CHECK-LABEL: @spam(
+; CHECK-NEXT:  bb:
+; CHECK-NEXT:    [[ALLOCA1:%.*]] = alloca [0 x [30 x %struct.widget]], align 16
+; CHECK-NEXT:    call void @zot(ptr nonnull [[ALLOCA1]])
+; CHECK-NEXT:    ret void
+;
+bb:
+  %alloca = alloca [30 x %struct.widget], i32 0, align 16
```

Comment:
Fix alloca of 0 size. Also all of these should use the alloca address space

Reply:
> Drop the triples
> Avoid using undef and use named values
> Drop CC and weak
> Fix alloca of 0 size. Also all of these should use the alloca address space

removed CC, weak and triples. updated alloca address space and used named values.